### DISCUSSION TITLE
[BP#59] Redimensionner les images en dehors du CMS

### DISCUSSION THREAD

**Discussion by Fred-artwai** [cnumr/best-practices#240] [Bonnes pratiques] [answered]
Comme pour la [BP#34] Redimensionner les images en dehors du HTML, je suis très dubitatif sur cette BP qui complètement en dehors de la réalité de mon terrain.

Cf la discussion sur la BP#34

Si l'objet de cette BP est d'épargner la bande passante alors je propose de donner une définir la BP comme suit : 
"Fixer une limite de taille et de poids pour les images uploadées"
Une image a rarement besoin de dépasser les 1920x1200 taille standard d'un écran de PC et un poids de 2 ou 3Mo est suffisant pour ce genre de fichier.

**Comment by ACTLEM** [comment]
L'idée ici est je pense d'éviter de redimensionner à chaud une image (ou du moins à chaque fois) côté serveur. Enfin, c'est comme cela que je le comprends. Elle complète donc la 34 (redimensionnement côté navigateur).

**Reply by Fred-artwai** [reply]
Généralement on dépose une image et chaque version de l'image est créée une fois et non pas à chaque fois.

**Reply by tbroyer** [reply]
Mon interprétation de cette BP a toujours été de faire les traitements de redimensionnement et optimisation avant d'uploader l'image.
Je ne suis pas persuadé que ce soit un levier important.

**Reply by florinesueur** [reply]
Je la comprends comme @tbroyer. Comment peut-on en mesurer l'impact ?
J'avais fait un test avec NetlifyCMS et gatsby avec sharp et ce fut une horreur en termes de temps de build.
Aussi une question, n'est pas intéressant en termes de vidéo ? Vaut-il mieux pas faire un petit coup de FFmpeg sur son ordinateur puis d'uploader la vidéo sur un cms ou réseau social ? Désolée, je sors de la BP.

**Reply by ACTLEM** [reply]
Effectivement en relisant plus attentivement la BP, @tbroyer a raison. Je l'ai lue trop vite. Merci pour la correction.
@Fred-artwai, nous sommes d'accord c'est normalement ce qui est fait (par exemple par défaut sur Wordpress). En réalité, j'ai été influencé par une expérience où l'image était traitée et retaillée en direct côté serveur (via un module apache de resizing).
Concernant la BP, le gain global semble effectivement très marginal finalement.

**Comment by ACTLEM** [comment, marked as answer]
Issue pour suppression: https://github.com/cnumr/best-practices/issues/265
